pygatt/bled112_backend.py

```diff
@@ -211,9 +216,9 @@ def bond(self):
         self._lib.send_command(self._ser, cmd)
 
         # Wait for response
-        self._loglock.release()  # don't hold loglock while waiting
-        self._wait_for_cmd_response()
-        self._loglock.acquire()
+        self._process_packets_until(
+            [self._lib.PacketType.ble_rsp_sm_set_bondable_mode,
+             self._lib.PacketType.ble_evt_connection_disconnected])
```

Comment:
We always should handle a disconnected event (an incoming packet will always have it's handler called if it appears before one of the expected packets in _process_packets_until). It doesn't make sense to handle it somewhere else. 

On a related note, I just realized that some of the functions that don't rely on a connection will stop at a disconnect event. I'll change that.